.github/workflows/python.yaml

```diff
@@ -72,4 +72,11 @@ jobs:
       with:
         version: ${{ env.CURRENT_VERSION }}
         alias: development
-        push: ${{ github.ref == 'refs/heads/main' }}  # build always, publish on 'main' only to prevent version clutter
+        push: ${{ github.ref == 'refs/heads/main' }} # build always, publish on 'main' only to prevent version clutter
+    - name: Execute jupytext in demo venv
+      run: |
+        python -m venv .demovenv
+        source .demovenv/bin/activate
+        pip install -r demos/requirements.txt
+        python demos/demo-ci/jupytext_to_executed_notebook.py ./demos/demo-ci/create_demo_notebook.py ./demos/demo_data_science_project.ipynb
+        rm -rf data .lakectl.yaml .demovenv
```

Comment:
As briefly discussed offline, while I find jupytext appealing, using the mkdocs `mknotebooks` plugin produces much nicer output that doesn't have a "notebook-y" vibe.

I'd be open to `mkdocs-jupyter` as a halfway-point, which can directly work with jupytext input, but relieves us of all the custom logic in the `create_demo_notebook.py` script.

Reply:
As it is right now, the notebook gets executed by the custom script and an `.ipynb` is generated. This we could then also use with `mknotebooks`. Or do you have an issue with the custom logic in the `jupytext_to_executed_notebook.py` script?

Reply:
Tagging @janwillemkl